plugins/catalog-react/src/components/EntityRefLink/EntityRefLink.tsx

```diff
@@ -39,6 +61,102 @@ export type EntityRefLinkProps = {
   children?: React.ReactNode;
 } & Omit<LinkProps, 'to'>;
 
+type PeekAheadPopoverProps = {
+  popupState: PopupState;
+  entityRef: CompoundEntityRef;
+};
+
+const useStyles = makeStyles(() => {
+  return {
+    popoverPaper: {
+      width: '20em',
+    },
+  };
+});
+
+export const PeekAheadPopover = ({
+  popupState,
+  entityRef,
+}: PeekAheadPopoverProps) => {
+  const entityRoute = useRouteRef(entityRouteRef);
+  const classes = useStyles();
+  const apiHolder = useApiHolder();
+  const [entity, setEntity] = useState<Entity | undefined>();
+
+  const { loading, error } = useAsync(async () => {
+    if (!entity) {
```

Comment:
I added this optimization to prevent it loading each time the popup is opened.

Reply:
Hm this one isn't quite right though, because of the async nature of things it'll happily retrigger loads multiple times in parallel since the setEntity happens some time later down the line.

What you should do instead is to leverage useAsyncFn to detach the async function from the trigger action. Something like

```tsx
const [{ loading, error, value }, load] = useAsyncFn(async () => {
  // do the loading
}, [apiHolder, entityRef]);

useEffect(() => {
  if (popupState.isOpen && !value && !error) {
    load();
  }
}, [popupState.isOpen, load, value]);
```

that's written entirely without testing, so you might want to validate carefully that it works :) especially if you insert a synthentic slow-down inside the useAsyncFn, like `await new Promise(r => setTimeout(r, 5000));` and then try to open and close the popop quickly while watching the network inspector

Reply:
btw a classic danger is if entityRef is an object, and that object gets reconstructed each and every time; then even though it's equivalent to last time in terms of its value, the ref will be different so it'll look like a change and trigger rerenders anyway